### ISSUE TITLE
Adding a column (if quantitative trait) or two (if binary trait) to the table ”Gene prioritisation using colocalisation analysis” on the Gene prioritisation page to infer directionality of effect

### ISSUE THREAD

**Issue by MayaGhoussaini** [opentargets/issues#2651]
These additional columns will help users to infer whether an increase or decrease in the expression of a given gene/ abundance of a given protein in a given tissue/cell type is likely to cause an increase in the trait/increase of disease risk.
 
The column headers could be “Direction of effect” for the quantitative trait phenotypes and “Direction of effect” and “Therapeutic mode of action based on GWAS/QTL” for the binary traits. We need to discuss this before implementation. 
 
 
There are 8 scenarios. 
 
For **Quantitative traits** 
 
Here, I'm referring to β1 as the β from the association summary (left study) and β2 as the QTL beta displayed in the “Gene prioritisation using colocalisation analysis” (right study) under the table tab. 
 
 
Scenario 1: β1 is positive and β2 is positive: ↑ expression ↑ trait
 
Scenario 2: β1 is negative and β2 is negative: ↓ expression ↓ trait 
 
Scenario 3: β1 is positive and β2 is negative: ↓ expression ↑ trait
                        
Scenario 4: β1 is negative and β2 is positive: ↑ expression ↓ trait
 
Writing this down makes me realise that we cannot infer therapeutic directionality for quantitative traits. Some proteins’ increased levels is protective while others increased level increases risk (i.e. HDL cholesterol and LDL cholesterol). 
 
For **Binary traits**
 
We could potentially have two columns added: “Direction of effect” and “Therapeutic mode of action based on GWAS/QTL”. However we need to be cautious about inferring therapeutic directionality from GWAS/QTLs. Also QTL directionality can vary sometimes between tissues.
 
Scenario 5: OR>1 and β2 is positive: ↑ expression ↑ disease risk -> Inhibition is therapeutic
 
Scenario 6: OR<1 and β2 is negative:  ↓expression ↓ disease risk -> Inhibition is therapeutic
 
 
Scenario 7: OR>1 and β2 is negative: ↓ expression ↑ disease risk -> Stimulation is therapeutic
 
Scenario 8: OR<1 and β2 is positive: ↑ expression ↓ disease risk -> Stimulation is therapeutic

**Comment by d0choa** [comment]
IMO, we should remain in the "direction of effect" concept and avoid the "therapeutic directionality" assessment.

Among other things, some traits like menopause or homosexuality might be problematic.

**Comment by ireneisdoomed** [comment]
Q: Is the type of trait somehow encoded in EFO or would it be an inference based on the beta/OR values?

**Comment by d0choa** [comment]
@MayaGhoussaini we should explore what's the uncertainty of the assessment. A few questions that would be good to investigate:

- Do we believe all h4/h3 values for this assessment?
- Do we want to include all effect sizes that marginally deviate from 1? 
- Should there be an intermediate case in which there is enough uncertainty on the data not to make the call?

I feel we could get these answers by comparing the assessments with other potentially cleaner sources of the direction of effect (e.g. drug data for binary traits or burden tests for quantitative traits)

@ireneisdoomed 
> Q: Is the type of trait somehow encoded in EFO or would it be an inference based on the beta/OR values?

To do it on the data, we need to start with the coloc dataset and join the left variant - left study beta.

```
> root
 |-- coloc_n_vars: integer (nullable = true)
 |-- coloc_h0: double (nullable = true)
 |-- coloc_h1: double (nullable = true)
 |-- coloc_h2: double (nullable = true)
 |-- coloc_h3: double (nullable = true)
 |-- coloc_h4: double (nullable = true)
 |-- left_type: string (nullable = true)
 |-- left_study: string (nullable = true)
 |-- left_chrom: string (nullable = true)
 |-- left_pos: integer (nullable = true)
 |-- left_ref: string (nullable = true)
 |-- left_alt: string (nullable = true)
 |-- right_type: string (nullable = true)
 |-- right_study: string (nullable = true)
 |-- right_bio_feature: string (nullable = true)
 |-- right_phenotype: string (nullable = true)
 |-- right_chrom: string (nullable = true)
 |-- right_pos: integer (nullable = true)
 |-- right_ref: string (nullable = true)
 |-- right_alt: string (nullable = true)
 |-- coloc_h4_h3: double (nullable = true)
 |-- coloc_log2_h4_h3: double (nullable = true)
 |-- is_flipped: boolean (nullable = true)
 |-- right_gene_id: string (nullable = true)
 |-- left_var_right_study_beta: double (nullable = true)
 |-- left_var_right_study_se: double (nullable = true)
 |-- left_var_right_study_pval: double (nullable = true)
 |-- left_var_right_isCC: boolean (nullable = true)
```

There is an alternative, do it in the front-end, but it might be a bit messy.

**Comment by MayaGhoussaini** [comment]
That is a good point David.  Yes agree. We should stick to the first column for everything. I was anyway not very comfortable with the second column. Below are answers to your questions: 

Do we believe all h4/h3 values for this assessment?
I think these are separate things. We are not inferring directionality from the coloc and it is testing a different hypothesis (i.e. whether the trait and the QTL are likely to share the same causal variant. I think as long as we have a GWAS-significant hit and a significant QTL p-value then we can infer this column. We should add a tool tip to say that this is inferred from QTL beta and trait beta/OR.
Do we want to include all effect sizes that marginally deviate from 1? We could include any hit that is GWAS significant and use the OR> or <1 to infer allele/risk direction. 
Should there be an intermediate case in which there is enough uncertainty on the data not to make the call? well I guess if the eQTL is not significant then you cannot infer a direction. I think we display in the portal only significant eQTL/pQTL, isn't it?

**Comment by d0choa** [comment]
This has been now implemented in the genetics portal evidence that features in Platform #2831 #2840. It will be available on 23.02
